pytest_splinter/plugin.py

```diff
@@ -308,7 +311,10 @@ def prepare_browser(parent):
             prepare_browser(parent)
 
         def make_screenshot_on_failure():
-            if splinter_make_screenshot_on_failure and request.node.splinter_failure:
+            if not splinter_make_screenshot_on_failure or not request.node.splinter_failure:
```

Comment:
please change back, it decreases the coverage, also the readability - those multiple returns in a single function